src/components/PieChart/index.jsx

```diff
@@ -0,0 +1,110 @@
+import React, { useEffect, useState } from "react";
+import { Pie } from "react-chartjs-2";
+import { generateRandomColors, createColorsWithOpacity } from "../../shared/utils";
+
+const CHART_COUNT = 8;
+
+const PieChart = ({ data }) => {
+    const colors = generateRandomColors(CHART_COUNT);
```

Comment:
this component looks like copy pasted bar chart. all code duplicates. think how you can have this code written once and reuse it instead of copy pasting

Reply:
and what do you suggest to do ?

Reply:
think more

Reply:
well